Transfer/Sources/TransferAmountCell.swift

```diff
@@ -0,0 +1,159 @@
+            .constraint(equalTo: detailTextLabel!.layoutMarginsGuide.leadingAnchor, constant: -15).isActive = true
+        let amountTextFieldLeadingConstraint = amountTextField.safeAreaLeadingAnchor
+            .constraint(equalTo: textLabel!.layoutMarginsGuide.trailingAnchor, constant: 15)
+        amountTextFieldLeadingConstraint.priority = UILayoutPriority(999)
+        amountTextFieldLeadingConstraint.isActive = true
+    }
+
+    func configure(amount: String?, currency: String?, isEnabled: Bool, _ handler: @escaping EnteredAmountHandler) {
+        textLabel?.text = "transfer_amount".localized()
+        amountTextField.text = amount
+        amountTextField.isEnabled = isEnabled
+        detailTextLabel?.text = currency
+        enteredAmountHandler = handler
+    }
+}
+
+extension TransferAmountCell: UITextFieldDelegate {
+    private static var amountFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.maximumFractionDigits = 2
+        formatter.minimumFractionDigits = 2
+        formatter.minimumIntegerDigits = 1
+        formatter.numberStyle = .currency
+        formatter.currencyCode = ""
+        formatter.currencySymbol = ""
+        formatter.locale = Locale(identifier: Locale.preferredLanguages[0])
+        return formatter
+    }()
+
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        enteredAmountHandler?(amountTextField.text ?? "")
+    }
+
+    func textField(_ textField: UITextField,
+                   shouldChangeCharactersIn range: NSRange,
+                   replacementString string: String) -> Bool {
+        setCursorPositionToTheEnd(of: textField)
+
+        var currentText = textField.text ?? ""
+        if string.count > 1 {
+            // Paste
+            guard let numberPastedAmount = TransferAmountCell.amountFormatter.number(from: string),
+                currentText.isEmpty else {
+                    return false
+            }
+            textField.text = TransferAmountCell.amountFormatter.string(from: numberPastedAmount)
+            return false
+        }
+        let digitsOnly = currentText.replacingOccurrences( of: "[^0-9]", with: "", options: .regularExpression)
+        if string.isEmpty {
+            // backspace
+            currentText = digitsOnly.dropLast().description
+        } else {
+            // digit
+            currentText = digitsOnly + string
+        }
+        if ["00", "0"].contains(currentText) {
+            textField.text = ""
+            return false
+        }
+        currentText = "0" + currentText
+        currentText.insert(".", at: currentText.index(currentText.endIndex, offsetBy: -2))
+        let numberAmount = TransferAmountCell.amountFormatter.number(from: currentText)!
+        textField.text = TransferAmountCell.amountFormatter.string(from: numberAmount)
+
+        return false
+    }
+
+    private func setCursorPositionToTheEnd(of textField: UITextField) {
```

Comment:
what is this function for?

Reply:
to set cursor to the end. User can tap on and change cursor position

Reply:
deleted